### PR TITLE
Tests - Add test cases for transformer block in GPT-like models

### PR DESCRIPTION
**Pull request by yzygitzh**
Add test cases for transformer block in GPT-like models to numerical tests. Including:
- Layer-norm (TE).
- RoPE.
- Attention (GQA).
- MLP.
- MoE layer.
- Bias-dropout-add (fused).

### CODE REVIEW

File: tests/numerical_tests/modules/test_bda.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    @pytest.mark.parametrize('steps', [10])
+    def test_bda(self, inputs_kv, config_kv, steps, request):
+        config = TransformerConfig(**config_kv)
+
+        if config.bf16:
+            dtype = torch.bfloat16
+        else:
+            dtype = torch.float32
+
+        model = build_module(get_bias_dropout_add)
```
[the remark this answers is not in the document]
Comment (yzygitzh) [reply tests/numerical_tests/modules/test_bda.py:40]: No, `build_module` handles function and module class at the same time.

File: tests/numerical_tests/modules/test_bda.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        },
+    ])
+    @pytest.mark.parametrize('config_kv', [
+        {
+            'use_cpu_initialization': True,
+            'bf16': True,
+            'num_layers': 1,
+            'num_attention_heads': 64,
+            'kv_channels': 128,
+            'hidden_dropout': 0.0,
```
[the remark this answers is not in the document]
Comment (yzygitzh) [reply tests/numerical_tests/modules/test_bda.py:27]: This is on purpose to align with current configuration.

File: tests/numerical_tests/modules/test_bda.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    inputs_kv['seq_length'],
+                    inputs_kv['micro_batch_size'],
+                    config.num_attention_heads,
+                    config.kv_channels,
+                ),
+                dtype=dtype,
+                requires_grad=True,
+            )
+            output = model(training=True, fused=config.bias_dropout_fusion)(
+                (mlp_input.cuda(), mlp_bias.cuda()), residual.cuda(), config.hidden_dropout
```
[the remark this answers is not in the document]
Comment (yzygitzh) [reply tests/numerical_tests/modules/test_bda.py:74]: `bias_droptou_fusion` has a default value in TransformerConfig class.

File: tests/numerical_tests/modules/test_rope.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    @pytest.mark.parametrize('inputs_kv', [
+        {
+            'seq_length': 4096,
+            'micro_batch_size': 1,
+        },
+    ])
+    @pytest.mark.parametrize('config_kv', [
+        {
+            'use_cpu_initialization': True,
+            'bf16': True,
```
Comment (cp5555) [inline tests/numerical_tests/modules/test_rope.py:24]: If pf16 is True, how to handle FP32?

File: tests/numerical_tests/modules/test_rope.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        else:
+            dtype = torch.float32
+
+        model = RotaryEmbedding(
+            kv_channels=config.kv_channels,
+            rotary_percent=rope_kv['rotary_percent'],
+        )
+        model = model.cuda()
+        if config.bf16:
+            model = Float16Module(config, model)
```
Comment (cp5555) [inline tests/numerical_tests/modules/test_rope.py:50]: Shall we write an utility function to load configuration for all test? I think current code are very similar. It's more like you have a default configuration for GPT-like model, and you can rewrite the configuration by config.

File: tests/numerical_tests/modules/test_moe_layer.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                experts=ModuleSpec(
+                    module=GroupedMLP,
+                    submodules=None
+                ),
+                shared_experts=ModuleSpec(
+                    module=SharedExpertMLP,
+                    params={"gate": False},
+                    submodules=MLPSubmodules(
+                        linear_fc1=TEColumnParallelLinear,
+                        linear_fc2=TERowParallelLinear,
```
Comment (cp5555) [inline tests/numerical_tests/modules/test_moe_layer.py:85]: It's more like we only test specific setting of MoE layer.

File: tests/numerical_tests/modules/test_mlp.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        if config.bf16:
+            dtype = torch.bfloat16
+        else:
+            dtype = torch.float32
+
+        module_spec = ModuleSpec(
+            module=MLP,
+            submodules=MLPSubmodules(
+                linear_fc1=TELayerNormColumnParallelLinear,
+                linear_fc2=TERowParallelLinear,
```
Comment (cp5555) [inline tests/numerical_tests/modules/test_mlp.py:46]: Why this setting?

**Review by abuccts — APPROVED** [review]
pls trigger the tests in pipeline scripts
